awslabs/aws-servicebroker
# feat: Add support for AWS CodeCommit

## Overview

Add support for CodeCommit service
Python 2.7 is quite old and may be deprecated soon, using new 3.7.

Introduce capabilities to manage codecommit service to service user
through setup cloudformation script.

## Related Issues

--

## Testing

I did use it with local S3 bucket to create and destroy service numerous times.

I have also used this Kubernetes manifest to test it :
```yaml
---
apiVersion: servicecatalog.k8s.io/v1beta1
kind: ServiceInstance
metadata:
  name: test-aws-service
spec:
  clusterServiceClassExternalName: codecommit
  clusterServicePlanExternalName: default
--- 
apiVersion: servicecatalog.k8s.io/v1beta1
kind: ServiceBinding
metadata:
  name: test-aws-service
spec:
  instanceRef:
    name: test-aws-service
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: test-service-scripts
data:
    test-git.sh: |
        REPOSITORY_PASSWORD=`echo -n "$REPOSITORY_PASSWORD" | sed 's#/#%2F#g'`
        echo $REPOSITORY_URL_HTTP | sed "s#https://#https://$REPOSITORY_USER_NAME:$REPOSITORY_PASSWORD@#" > ~/.git-credentials
        set -x
        set -e
        apk -U --no-cache add git
        git config --global credential.helper store
        git clone $REPOSITORY_URL_HTTP $REPOSITORY_NAME
        cd $REPOSITORY_NAME
        git config --global user.email "you@example.com"
        git config --global user.name "Example User"
        date -R > test-date
        git add test-date
        git commit -m "test"
        git push origin master:master
        exec /bin/sleep 3650d
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: sleep
spec:
  replicas: 1
  selector:
    matchLabels:
      app: sleep
  template:
    metadata:
      labels:
        app: sleep
    spec:
      containers:
      - name: sleep
        image: alpine:edge
        command: ["/bin/sh", "/opt/scripts/test-git.sh"]
        imagePullPolicy: IfNotPresent
        env:
          - name: REPOSITORY_NAME
            valueFrom:
              secretKeyRef:
                name: test-aws-service
                key: REPOSITORY_NAME
          - name: REPOSITORY_URL_HTTP
            valueFrom:
              secretKeyRef:
                name: test-aws-service
                key: REPOSITORY_URL_HTTP
          - name: REPOSITORY_PASSWORD
            valueFrom:
              secretKeyRef:
                name: test-aws-service
                key: REPOSITORY_PASSWORD
          - name: REPOSITORY_USER_NAME
            valueFrom:
              secretKeyRef:
                name: test-aws-service
                key: REPOSITORY_USER_NAME
        volumeMounts:
        - name: scripts-volume
          mountPath: /opt/scripts
          readOnly: true
      volumes:
      - name: scripts-volume
        configMap:
          name: test-service-scripts
```

### Notes

--
## Testing Instructions

* Put the template to use (either in dev branch on central bucket, or in your custom bucket)
* Put the above manifest to your kubernetes cluster
* Wait till the pod is running, and check out the logs using `kubectl logs <pod_name>`

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
